File: 1646-kth-missing-positive-number/kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int n=arr.length;
       int max=arr[n-1];
       int count=0;
       List<Integer>list=new ArrayList<>();
       for(int x:arr)list.add(x);
       for(int i=1;i<=max+1;i++){
        if(!list.contains(i)){
            count++;
            if(count==k){
                return i;
            }
        }
       }
       if(count==1)return arr[n-1]+k;
       return arr[n-1]+k-count+1;
    }
}

File: 1646-kth-missing-positive-number/kth-missing-positive-number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int l = 0, r = arr.size();  // Binary search range
        while (l < r) {
            int mid = l + (r - l) / 2;
            int missingCount = arr[mid] - (mid + 1);  // How many numbers are missing before arr[mid]
            if (missingCount < k) {
                l = mid + 1;  // Move right
            } else {
                r = mid;  // Move left
            }
        }
        // The kth missing number is l + k
        return r + k;
    }
};